docs/code-examples/repository-pattern/01_simple_example/ilGeoLocationRepository.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+class ilGeoLocationRepository {
+
+    public const $table = 'geo_location';
+
+    public function __construct(ilDB $a_db)
+    {
+        $this->db = $a_db;
+    }
+
+    public function createGeoLocation(ilObjGeoLocation $obj)
+    {
+        $this->db->insert($this->table, array(
+            'id' => array('integer', $obj->getId()),
+            'title' => array('text', $obj->getTitle()),
+            'lattitude' => array('float', $obj->getLattitude()),
+            'longitude' => array('float', $obj->getLongitude()),
+            'expiration_timestamp' => array('timestamp', $obj->getIExpirationTimestamp())
+        ));
+    }
+
+    public function getGeoLocationById($a_id)
+    {
+        $query = "Select * FROM $this->table WHERE id = " . $this->db->quote($a_id, 'integer');
+        $result = $this->db->query($query);
+
+        if($row = $this->db->fetchAssoc($result))
+        {
+            return new ilObjGeoLocation($row['id'], $row['title'], $row['lattitude'], $row['longitude'], $row['expiration_timestamp'])
+        }
+        else
+        {
+            return NULL;
+        }
+    }
+
+    public function getGeoLocationsByCoordinates($a_lattitude, $a_longitude)
+    {
+        $query = "Select * FROM $this->table WHERE lattitude = " . $this->db->quote($a_lattitude, 'float')

```

Comment:
I would suggest to quote the identifier of the table as well. I think `ilDB` has a method for that.

Reply:
I would happily quote every time and consider this a mikro optimization without much effect overall, until proven otherwise.